File: leetcode/995.minimum-number-of-k-consecutive-bit-flips.js
/**
 * @param {number[]} A
 * @param {number} K
 * @return {number}
 */
var minKBitFlips = function(A, K) {
  let cnt = 0
  for (let i = 0; i <= A.length - K; i++) {
    if (A[i] === 0) {
      cnt++
      for (let j = i; j < i + K; j++) {
        A[j] = A[j] ^ 1
      }
      // console.log(A)
    }
  }
  for (let i = A.length - K; i < A.length; i++) {
    if (A[i] === 0) {
      return -1
    }
  }
  return cnt
}

File: leetcode/995.minimum-number-of-k-consecutive-bit-flips.go
func flip(arr []int, K int, from int)  {
	for i := from; i < from + K; i++ {
		arr[i] ^= 1
	}
}

func minKBitFlips(A []int, K int) int {
	cnt := 0
	l := len(A)
	for idx, v := range A {
		if v == 0 {
			if idx + K > l {
				return -1
			}
			flip(A, K, idx)
			// fmt.Println(A)
			cnt++
		}
	}
	return cnt
}